azure/services/securitygroups/securitygroups.go

```diff
@@ -18,142 +18,96 @@ package securitygroups
+		for _, nsgSpec := range s.Scope.NSGSpecs() {
+			if _, err := s.CreateResource(ctx, nsgSpec, serviceName); err != nil {
+				if !azure.IsOperationNotDoneError(err) || resErr == nil {
+					resErr = err
 				}
 			}
-			if !update {
-				// Skip update for NSG as the required default rules are present
-				log.V(2).Info("security group exists and no default rules are missing, skipping update", "security group", nsgSpec.Name)
-				continue
-			}
-		default:
-			log.V(2).Info("creating security group", "security group", nsgSpec.Name)
-			for _, rule := range nsgSpec.SecurityRules {
-				securityRules = append(securityRules, converters.SecurityRuleToSDK(rule))
-			}
 		}
-		sg := network.SecurityGroup{
-			Location: to.StringPtr(s.Scope.Location()),
-			SecurityGroupPropertiesFormat: &network.SecurityGroupPropertiesFormat{
-				SecurityRules: &securityRules,
-			},
-			Etag: etag,
-		}
-		err = s.client.CreateOrUpdate(ctx, s.Scope.ResourceGroup(), nsgSpec.Name, sg)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create or update security group %s in resource group %s", nsgSpec.Name, s.Scope.ResourceGroup())
-		}
-
-		log.V(2).Info("successfully created or updated security group", "security group", nsgSpec.Name)
 	}
-	return nil
-}
 
-func ruleExists(rules []network.SecurityRule, rule network.SecurityRule) bool {
-	for _, existingRule := range rules {
-		if !strings.EqualFold(to.String(existingRule.Name), to.String(rule.Name)) {
-			continue
-		}
-		if !strings.EqualFold(to.String(existingRule.DestinationPortRange), to.String(rule.DestinationPortRange)) {
-			continue
-		}
-		if existingRule.Protocol != network.SecurityRuleProtocolTCP &&
-			existingRule.Access != network.SecurityRuleAccessAllow &&
-			existingRule.Direction != network.SecurityRuleDirectionInbound {
-			continue
-		}
-		if !strings.EqualFold(to.String(existingRule.SourcePortRange), "*") &&
-			!strings.EqualFold(to.String(existingRule.SourceAddressPrefix), "*") &&
-			!strings.EqualFold(to.String(existingRule.DestinationAddressPrefix), "*") {
-			continue
-		}
-		return true
-	}
-	return false
+	s.Scope.UpdatePutStatus(infrav1.SecurityGroupsReadyCondition, serviceName, resErr)
+	return resErr
 }
 
-// Delete deletes the network security group with the provided name.
+// Delete deletes network security groups.
 func (s *Service) Delete(ctx context.Context) error {
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "securitygroups.Service.Delete")
 	defer done()
 
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
+
+	// Only delete the NSG if its lifecycle is managed by this controller.
 	if !s.Scope.IsVnetManaged() {
-		log.V(4).Info("Skipping network security group delete in custom VNet mode")
+		log.V(4).Info("Skipping network security groups delete in custom VNet mode")
```

Comment:
It looks like the log level for skipping NSG reconcile is a `log.Info()` while this is a `log.V(4).Info()`, should they be the same log level?

Reply:
changed both to V(4) since route tables are also using V(4)